src/NuGet.Core/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -451,6 +494,41 @@ public void Close()
             }
         }
 
+        /// <summary>
+        /// Write the XDocument to the packages.config and disallow further changes.
+        /// </summary>
+        /// <param name="fullPath">the full path to packages.config file</param>
+        public void Close(string fullPath)
+        {
+            var directorypath = Path.GetDirectoryName(fullPath);
+
+            var tempConfigPath = Path.Combine(directorypath, Path.GetRandomFileName());
```

Comment:
use a better name here for the temp file. It could be packages.config.{newguid} for example, so it is easy to understand how to recover if something went wrong